src/test/credentials/provider/ssoAccessTokenProvider.test.ts

```diff
@@ -0,0 +1,230 @@
+
+            stubCreateToken.onSecondCall().returns(({
+                promise: sandbox.stub().resolves(fakeCreateTokenResponse),
+            } as any) as AWS.Request<CreateTokenResponse, AWS.AWSError>)
+
+            const stubSaveAccessToken = sandbox.stub(cache, 'saveAccessToken').returns()
+
+            const startTime = Date.now()
+            const receivedToken = await sut.accessToken()
+            const endTime = Date.now()
+
+            const durationInSeconds = (endTime - startTime) / 1000
+
+            assert.strictEqual(durationInSeconds > 1, true)
+
+            assert.strictEqual(receivedToken.startUrl, validAccessToken.startUrl)
+            assert.strictEqual(receivedToken.region, validAccessToken.region)
+            assert.strictEqual(receivedToken.accessToken, validAccessToken.accessToken)
+
+            assert.strictEqual(stubSaveAccessToken.calledOnce, true)
+        })
+
+        it('should stop polling for unspecified errors during createToken call', async () => {
+            setUpStubCache(validRegistation)
+            sandbox.stub(sut, 'authorizeClient').resolves(validAuthorization)
+
+            const stubCreateToken = sandbox.stub(ssoOidcClient, 'createToken')
+            stubCreateToken.returns(({
+                promise: sandbox.stub().throws({ code: 'ErrorNotSlowDownAuthPendingOrTimeOut' }),
+            } as any) as AWS.Request<CreateTokenResponse, AWS.AWSError>)
+
+            const stubSaveAccessToken = sandbox.stub(cache, 'saveAccessToken').returns()
+
+            try {
+                await sut.accessToken()
+            } catch (err) {
+                assert.strictEqual(stubCreateToken.callCount, 1)
+            }
+
+            assert.strictEqual(stubSaveAccessToken.called, false)
+        })
+
+        it('should add backoff dely on SlowDownException', async () => {
+            setUpStubCache(validRegistation)
+            sandbox.stub(sut, 'authorizeClient').resolves(validAuthorization)
+
+            const stubCreateToken = sandbox.stub(ssoOidcClient, 'createToken')
+            stubCreateToken.onFirstCall().returns(({
+                promise: sandbox.stub().throws({ code: 'SlowDownException' }),
+            } as any) as AWS.Request<CreateTokenResponse, AWS.AWSError>)
+
+            stubCreateToken.onSecondCall().returns(({
+                promise: sandbox.stub().resolves(fakeCreateTokenResponse),
+            } as any) as AWS.Request<CreateTokenResponse, AWS.AWSError>)
+
+            const stubSaveAccessToken = sandbox.stub(cache, 'saveAccessToken').returns()
+
+            const startTime = Date.now()
+            const returnedToken = await sut.accessToken()
+            const endTime = Date.now()
+
+            const durationInSeconds = (endTime - startTime) / 1000
+
+            //The default backoff delay is 5 seconds, the starting retry interval is 1 second
+            assert.strictEqual(durationInSeconds > 6, true, 'Duration not over 6 seconds')
```

Comment:
Your duration related tests almost double the time it takes for the test suite to run. Where possible, we try to avoid using the real time clock so tests run as fast as possible. Search for other tests that use `lolex` for examples of tests that mock the system time.

Reply:
Implemented a lolex clock.